pandas/core/arrays/datetimelike.py

```diff
@@ -327,6 +327,10 @@ def __getitem__(self, key):
                              "numpy.newaxis (`None`) and integer or boolean "
                              "arrays are valid indices")
 
+        if key is Ellipsis:
```

Comment:
can you make a more explicit comment. This is likely a more general fix (IOW this is likely broken for all ``__getitem__`` on EA)

Reply:
why don't this just work as is? why is this not a view (and not a copy)?

Reply:
> why don't this just work as is?

as-is `DatetimeIndex[...]` loses its `freq` attribute

>why is this not a view (and not a copy)?

Since the copy doesn't have `deep=True`, `self.copy()` _effectively_ a view?

Reply:
can you be more explicit in your comment, this is not obvious

Reply:
Moved to a hopefully clearer spot a few lines down